src/main/java/net/minecraftforge/network/ForgeNetwork.java

```diff
@@ -0,0 +1,283 @@
+     * Sends the specified entity's equipment caps to all tracking entities
+     * including itself in the case of it being a player.
+     * 
+     * @param livingEntity      - the entity for whom's equipment caps you're
+     *                          sending
+     * @param equipmentSlotType - the equipment slot
+     * @param itemStack         - the equipment item stack
+     * @param writeAll          - <code>true</code> to write all cap data,
+     *                          <code>false</code> to write only dirty data.
+     */
+    public static void sendEquipmentSlotCapabilities(LivingEntity livingEntity, EquipmentSlot equipmentSlotType,
+        ItemStack itemStack, boolean writeAll)
+    {
+        sendEquipmentSlotCapabilities(livingEntity, equipmentSlotType, itemStack, writeAll, true,
+            getTrackingConnections(livingEntity));
+    }
+
+    /**
+     * Sends the specified entity's equipment caps to a set amount of players.
+     * 
+     * @param livingEntity      - the entity for whom's equipment caps you're
+     *                          sending
+     * @param equipmentSlotType - the equipment slot
+     * @param itemStack         - the equipment item stack
+     * @param writeAll          - <code>true</code> to write all cap data,
+     *                          <code>false</code> to write only dirty data.
+     * @param sendToSelf        - send to itself (if the entity is a player)
+     * @param players           - the players to send the entity's equipment caps to
+     */
+    public static void sendEquipmentSlotCapabilities(LivingEntity livingEntity, EquipmentSlot equipmentSlotType,
+        ItemStack itemStack, boolean writeAll, boolean sendToSelf, Stream<Connection> players)
+    {
+        if (!livingEntity.level.isClientSide() && (writeAll || itemStack.requiresSync()))
+        {
+            FriendlyByteBuf capabilityData = new FriendlyByteBuf(Unpooled.buffer());
+            itemStack.encode(capabilityData, writeAll);
+            if (!capabilityData.isReadable())
+            {
+                return;
+            }
+            Packet<?> packet = CHANNEL.toVanillaPacket(
+                new EquipmentSlotCapabilitiesMessage(livingEntity.getId(), equipmentSlotType, capabilityData),
+                NetworkDirection.PLAY_TO_CLIENT);
+            if (sendToSelf && livingEntity instanceof ServerPlayer)
+            {
+                sendIfPresent(packet, ((ServerPlayer) livingEntity).connection.connection);
+            }
+            broadcast(packet, players);
+        }
+    }
+
+    public static Stream<Connection> getTrackingConnections(Entity entity)
+    {
+        if (entity.level instanceof ServerLevel level)
+        {
+            ServerChunkCache chunkSource = level.getChunkSource();
+            Set<ServerPlayerConnection> players = chunkSource.chunkMap.getSeenBy(entity);
+            if (players != null)
+            {
+                return chunkSource.chunkMap.getSeenBy(entity).stream()
```

Comment:
You can directly use players here, it would increase perfs and readability.

Reply:
Whoops, didn't see that